### PR TITLE
[WIP] Fix ATen Fallback export mode for non-Caffe2 builds, plus refactor for consistency and debugability

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This PR 1) unifies `onnx::ATen` and `aten::ATen` nodes onto `aten::ATen` for consistency. ONNX doesn't have ATen op but PyTorch converter emits it by accident.

It also 2) Renames ATen's attribute `operator` to `operator_name` to allow access from C++ code. `operator` is a reserved word in C++ and any attempt to using `attr::operator` will raise a compiling error.

Although this PR does not use `attr::operator_name` directly, it enables users to do so during debug. For example, one developer debugging `onnx::operator` export could do something like

```
if (node->kind() == aten::ATen) {
    std::cout << "ATen operator name: " << node->s(::c10::attr::operator_name) << std::endl;
    std::cout << "ATen overload name: " << node->s(::c10::attr::overload_name) << std::endl;
}
```

Note that doing

```
if (node->kind() == aten::ATen) {
    std::cout << "ATen operator name: " << node->s(::c10::attr::operator) << std::endl;
}
```
would be invalid as C++ has reserved `operator` word.

### CODE REVIEW

**Review by BowenBao — COMMENTED** [review]
if `Renamed ATen's operator to operator_name to allow access from C++ code. operator is a reserved word and using attr::operator is not allowed.` is the only reason for the change, I think there are ways to workaround that. e.g. `Symbol::attr("operator")`.

File: aten/src/ATen/core/interned_strings.h [hunk 1]
```diff
@@ -496,7 +496,9 @@ namespace c10 {
   _(attr, cache_id)                  \
   _(attr, new_axis)                  \
   _(attr, warn_id)                   \
-  _(attr, allowzero)
+  _(attr, allowzero)                 \
+  _(attr, operator_name)             \
```
[the remark this answers is not in the document]
Comment (thiagocrepaldi) [reply aten/src/ATen/core/interned_strings.h:500]: What do you mean? I did log `attr::operator` during debug, but remove them before pushing the PR.

**Comment by thiagocrepaldi** [comment]
> if `Renamed ATen's operator to operator_name to allow access from C++ code. operator is a reserved word and using attr::operator is not allowed.` is the only reason for the change, I think there are ways to workaround that. e.g. `Symbol::attr("operator")`.

Wouldn't be easier to support both ways as it is in all other attributes? Having exception cause confusion

File: aten/src/ATen/core/interned_strings.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (BowenBao) [reply aten/src/ATen/core/interned_strings.h:500]: what is the use case? have not seen any code in this PR calling `attr::operator` hence the question.

**Comment by BowenBao** [comment]
> Wouldn't be easier to support both ways as it is in all other attributes? Having exception cause confusion

I don't object this change, just I don't know how often it would be used since I have not seen a use case in this PR. This change would potentially trigger some Meta internal test issues.

**Comment by BowenBao** [comment]
friendly ping for updates ... should we proceed with this PR?
If so please provide the example use case driving the change for renaming attribute `operator` to `operator_name` as discussed.

**Review by BowenBao — COMMENTED** [review]
Please resolve CI failed checks, otherwise LGTM.

**Comment by thiagocrepaldi** [comment]
@BowenBao It seems the current CI failures are unrelated to this change

**Comment by thiagocrepaldi** [comment]
Better merge after https://github.com/pytorch/pytorch/pull/73280

**Pull request closed without merging**